scapy/layers/dns.py

```diff
@@ -194,16 +194,18 @@ def h2i(self, pkt, x):
         return x
 
     def i2m(self, pkt, x):
-        if any((orb(y) >= 0xc0) for y in x):
-            # The value has already been processed. Do not process it again
-            return x
-
         if not x or x == b".":
             return b"\x00"
 
+        built = b"." not in x
+        built &= ((x[-1] == b"\x00") or ((orb(x[-2]) & 0xc0)) != 0)
```

Comment:
Why `& 0xc0` instead of `>= 0xc0` like it was before? If `&` is used, at least it should be `== 0xc0` instead of `!= 0`.

Reply:
Obviously. My bad :smile: I thought for one second I was dealing with 0x80